packages/subiquity_client/lib/src/server.dart

```diff
@@ -1,58 +1,49 @@
 import 'dart:io';
 
 import 'package:meta/meta.dart';
-import 'package:path/path.dart' as p;
-import 'package:xdg_directories/xdg_directories.dart' as xdg;
 
 import 'endpoint.dart';
 import 'server/common.dart';
+import 'server/launcher.dart';
 import 'server/paths.dart';
 
-abstract class SubiquityServer {
-  Process? _serverProcess;
-
-  SubiquityServer._();
-
-  /// Creates a new subiquity server.
-  factory SubiquityServer() => _SubiquityServerImpl();
-
-  /// Creates a WSL variant of the server.
-  factory SubiquityServer.wsl() => _WslSubiquityServerImpl();
+class SubiquityServer {
+  /// An optional server launcher, should we need to start the server.
+  SubiquityLauncher? launcher;
+
+  SubiquityServer._({this.launcher});
+
+  /// A variant of the server for Ubuntu Desktop Installer. By default starts
+  /// subiquity in dry-run. Set [isLiveRun] for a live run.
+  factory SubiquityServer({bool isLiveRun = false}) => SubiquityServer._(
+        launcher: isLiveRun
+            ? null
+            : SubiquityLauncher.python(
```

Comment:
can we construct the launcher in `udi/installer.dart`?

Reply:
Done. Async main helped a lot.